src/riakc_ts_query_operator.erl

```diff
@@ -31,9 +31,12 @@
 -export([serialize/2,
          deserialize/1, deserialize/2]).
 
-serialize(QueryText, Interpolations) ->
+
+serialize(QueryText, Interpolations)
+  when is_binary(QueryText) orelse is_list(QueryText) ->
+    Q = unicode:characters_to_binary(QueryText),
```

Comment:
When, at line 37, `Q` is assigned a term that is not of type binary(), it gets silently passed as field `base` in the `#tsinterpolation{}` record. The breakage will happen some time later, in the bowels of riakc_pb_socket (or in riak_pb), with a screenful of serialized multistorey stack trace, and with no mention of unicode conversion as a real cause. Importantly, the indication of the line where to look for offending code will be totally unhelpful. This all applies if we regard the bad unicode issue as a coding error on the part of client application (which calls `riakc_ts:query`), and not a "bad user input" error.

Perhaps if you check to make sure `Q` is not a tuple, in this function, the reason would be easier to identify.